iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -105,13 +105,18 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
           .executeWith(tableExecutor)
           .run(output -> {
             Table table = HiveIcebergStorageHandler.table(context.getJobConf(), output);
-            HiveIcebergRecordWriter writer = writers.get(output);
-            DataFile[] closedFiles = writer != null ? writer.dataFiles() : new DataFile[0];
-            String fileForCommitLocation = generateFileForCommitLocation(table.location(), jobConf,
-                attemptID.getJobID(), attemptID.getTaskID().getId());
-
-            // Creating the file containing the data files generated by this task for this table
-            createFileForCommit(closedFiles, fileForCommitLocation, table.io());
+            if (table != null) {
```

Comment:
Could we do this check in `HiveIcebergMetaHook.commitInsertTable`?
Then we would not need any change here...

Reply:
This happens during task commit, so before the commitInsert hook is called. 

The essential problem here is that `OUTPUT_TABLES` contains all the tables, however, only those tables are serialized into the jobconfig that are relevant for the given task. So it tries to iterate over 1...N tables, but only has access to serialized Table 1 (hence the if).

Reply:
Got it, then we should push this change to the Iceberg code as well.

Reply:
Sure, will do!

Reply:
Actually, this issue does not occur with mr, only with tez, therefore it currently does not come up in upstream Iceberg. Will fix this once Tez writes have been enabled there too.

Reply:
I think this is the same situation that we have been with some other Tez patches. It does not hurt to be there an we keep the two codes from diverging. But it's up to you to decide.

Reply:
Yeah, I think you're right